File: 02-programacion-basica/05-condicionales/scala/ejemplos-guiados/CondicionalesBasicos.scala
object CondicionalesBasicos:
  def main(args: Array[String]): Unit =
    val nota = 15
    val asistencia = 85

    // En Scala, if puede producir un valor y asignarse directamente.
    val estado =
      if nota < 0 || nota > 20 || asistencia < 0 || asistencia > 100 then
        "Datos inválidos"
      else if nota >= 11 && asistencia >= 70 then
        "Aprobado"
      else
        "Desaprobado"

    println(estado)

File: 02-programacion-basica/05-condicionales/java/ejemplos-guiados/CondicionalesBasicos.java
public class CondicionalesBasicos {
    public static void main(String[] args) {
        int nota = 15;
        int asistencia = 85;

        // Primero se valida el dominio de los datos para no clasificar entradas inválidas.
        if (nota < 0 || nota > 20 || asistencia < 0 || asistencia > 100) {
            System.out.println("Datos inválidos");
        } else if (nota >= 11 && asistencia >= 70) {
            System.out.println("Aprobado");
        } else {
            System.out.println("Desaprobado");
        }
    }
}
